src/phoenix/utilities/re.py

```diff
@@ -0,0 +1,50 @@
+from logging import getLogger
+from re import compile, split
+from typing import Dict, List, Mapping
+from urllib.parse import unquote
+
+_logger = getLogger(__name__)
+
+# Optional whitespace
+_OWS = r"[ \t]*"
+# A key contains printable US-ASCII characters except: SP and "(),/:;<=>?@[\]{}
+_KEY_FORMAT = r"[\x21\x23-\x27\x2a\x2b\x2d\x2e\x30-\x39\x41-\x5a\x5e-\x7a\x7c\x7e]+"
+# A value contains a URL-encoded UTF-8 string. The encoded form can contain any
+# printable US-ASCII characters (0x20-0x7f) other than SP, DEL, and ",;/
+_VALUE_FORMAT = r"[\x21\x23-\x2b\x2d-\x3a\x3c-\x5b\x5d-\x7e]*"
+# A key-value is key=value, with optional whitespace surrounding key and value
+_KEY_VALUE_FORMAT = rf"{_OWS}{_KEY_FORMAT}{_OWS}={_OWS}{_VALUE_FORMAT}{_OWS}"
+
+_HEADER_PATTERN = compile(_KEY_VALUE_FORMAT)
+_DELIMITER_PATTERN = compile(r"[ \t]*,[ \t]*")
+
+
+def parse_env_headers(s: str) -> Mapping[str, str]:
```

Comment:
```suggestion
def parse_env_headers(s: str) -> Dict[str, str]:
```

Suggestion: Make the return type as specific as possible.

Reply:
Hmm, I might not understand what this actually catches. I copied the type from OTEL. Will change but I might not understand the actual need for this type narrowing per-se.